root/static/scripts/edit/URLCleanup.js

```diff
@@ -2966,7 +2966,10 @@ const CLEANUPS: CleanupEntries = {
     match: [new RegExp('^(?:https?://)?(?:www\\.)?junodownload\\.com', 'i')],
     restrict: [LINK_TYPES.downloadpurchase],
     clean: function (url) {
-      return url.replace(/^https?:\/\/(?:www\.)?junodownload\.com\/([^?#]+).*$/, 'https://www.junodownload.com/$1');
+      url = url.replace(/^https?:\/\/(?:www\.)?junodownload\.com\/([^?#]+).*$/, 'https://www.junodownload.com/$1');
+      url = url.replace(/^https:\/\/www\.junodownload\.com\/(artists|labels)\/([^\/]+).*$/, 'https://www.junodownload.com/$1/$2/');
+      url = url.replace(/^https:\/\/www\.junodownload\.com\/products\/([^\/]+)\/([^\/]+).*$/, 'https://www.junodownload.com/products/$1/$2/');
```

Comment:
It doesn’t handle two existing variants of product path:
1. missing a slug name: `/products/1239234-02/`
1. with `.htm` file extension: `/products/1225060-02.htm`

Reply:
Hmm. Those styles both redirect to the page with the slug (https://www.junodownload.com/products/deepsky-jes-ghost/1239234-02/ for the first, the second is gone but the same link comes back if I add `.htm` to the first). We cannot clean these up to the slug page though, maybe we should clean all `/products` pages to remove the slug instead?

Reply:
That would work too.